pages/dkp/kommander/1.4/clusters/tunnel-cli/index.md

```diff
@@ -126,6 +126,54 @@ done
 kubectl get secret -n ${namespace} ${manifest} -o jsonpath='{.data.manifests\.yaml}' | base64 -d > manifest.yaml
 ```
 
+### Create a network policy for the tunnel server
+
+This step is optional, but improves security by restricting which remote hosts can connect to the tunnel.
+
+Apply a network policy that restricts the tunnel to specific IP blocks. The following example only permits connections from hosts with IP addresses in the ranges 192.0.2.0 to 192.0.2.255 and 203.0.113.0 to 203.0.113.255:
+
+```shell
+cat > net.yaml <<EOF
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  namespace: ${namespace}
+  name: ${name_connector}-deny
+  labels:
+    kubetunnel.d2iq.io/tunnel-connector: ${name_connector}
+    kubetunnel.d2iq.io/networkpolicy-type: "tunnel-server"
+spec:
+  podSelector:
+    matchLabels:
+      kubetunnel.d2iq.io/tunnel-connector: ${name_connector}
+  policyTypes:
+  - Ingress
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  namespace: ${namespace}
+  name: ${name_connector}-allow
+  labels:
+    kubetunnel.d2iq.io/tunnel-connector: ${name_connector}
+    kubetunnel.d2iq.io/networkpolicy-type: "tunnel-server"
+spec:
+  podSelector:
+    matchLabels:
+      kubetunnel.d2iq.io/tunnel-connector: ${name_connector}
+  policyTypes:
+  - Ingress
+  ingress:
```

Comment:
It looks good to me to use ipBlock as an example, but perhaps we want to restrict the access to a list of namespaces so one tunnel cannot see other tunnels. That is why I explained in the examples the iusage of the namespace selectors using `kubeaddons, kommander, kommander-system, ...`

Reply:
I had them in originally, but removing them did not break anything in terms of the tunnel working. Is there a difference we can see when they are set or not?

Reply:
When they are set only pods from those namespaces should be able to talk to the proxy server of tunnel A, while a namespace Foo isn't allowed.

Reply:
With the current configuration of deny-all with some IP address exceptions, pods from other namespaces will be denied. Why would we allow pods in `kubeaddons`, etc. to connect to these endpoints that are provided for agents to connect?

Reply:
For instance, we need our central thanos in kubeaddons ns to talk to the proxy-server to fetch the metrics using the tunnel. But this can apply to other namespaces such as kommander.